### PR TITLE
AAE-475 Apply variable mapping on StartEvent

### PR DESCRIPTION
**Pull request by CTI777**
Apply variable mapping on StartEvent

### CODE REVIEW

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/util/ProcessInstanceHelper.java [hunk 1]
```diff
@@ -129,11 +129,13 @@ else if (bpmnModel.containsMessageId(messageRef)) {
       throw new ActivitiException("No message start event found for process definition " + processDefinition.getId() + " and message name " + messageName);
     }
 
-    // Map message payload variables before creating process instance
-    Map<String, Object> processVariables = commandContext.getProcessEngineConfiguration()
-                                                         .getEventSubscriptionPayloadMappingProvider()
-                                                         .apply(messageVariables,
-                                                                eventSubscription);
+//    // Map message payload variables before creating process instance
+//    Map<String, Object> processVariables = commandContext.getProcessEngineConfiguration()
```
Comment (erdemedeiros) [inline activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/util/ProcessInstanceHelper.java]: If this code is not necessary anymore it should be completely removed. I guess that this is removing support for variable mappings for messages when declared inside the XML itself, but maybe it's not a big deal, as all the variable mappings are done using the extension files, right now. @igdianov do you any concern about that?

File: activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessExtensionsTest.java [hunk 2]
```diff
@@ -150,4 +151,36 @@ public void processInstanceFailsIfVariableTypeIncorrect() {
...
+        .isNotNull()
+        .hasSize(2)
+        .extracting(VariableInstance::getName,
+                    VariableInstance::getValue)
+        .containsOnly(
+                      tuple("name", "name_value"),
+                      tuple("email", "email_value")
+        );
+
+        processRuntime.delete(ProcessPayloadBuilder.delete(process));
```
Comment (erdemedeiros) [inline activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessExtensionsTest.java]: You can skip the process clean up here as the processes are already being deleted in `@After` method.

Comment (CTI777) [reply activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessExtensionsTest.java]: Done

File: activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/util/ProcessInstanceHelper.java (hunk 1 above)
Comment (CTI777) [reply activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/util/ProcessInstanceHelper.java]: Done

**Pull request merged**